python_modules/dagster/dagster/_core/definitions/graph_definition.py

```diff
@@ -746,27 +758,72 @@ def get_resource_requirements(
     @public
     @property
     def name(self) -> str:
+        """The name of the graph.
+        """
         return super(GraphDefinition, self).name
 
     @public
     @property
     def tags(self) -> Mapping[str, str]:
+        """The tags associated with the graph.
+        """
         return super(GraphDefinition, self).tags
 
     @public
     def alias(self, name: str) -> "PendingNodeInvocation":
+        """Aliases the graph with a new name.
+
+        Can only be used in the context of a :py:func:`@graph <graph>`, :py:func:`@job <job>`, or :py:func:`@asset_graph <asset_graph>` decorated function.
+
+        Examples:
+            .. code-block:: python
+            @job
+            def do_it_all():
+                my_graph.alias("my_graph_alias")
+
+        """
         return super(GraphDefinition, self).alias(name)
 
     @public
     def tag(self, tags: Optional[Mapping[str, str]]) -> "PendingNodeInvocation":
+        """Attaches the provided tags to the graph immutably.
```

Comment:
Tangential to this PR but it's pretty rough that `tag` and `tags` are so different. Seems like this should be named `with_tag` or something.

Reply:
I agree :/